content/blog/greenlake-platform-infrastructure-monitoring-with-apache-skywalking.md

```diff
@@ -0,0 +1,138 @@
+
+## **Pre-requisites**
+
+* An active service subscription to HPE GreenLake Private Cloud Enterprise.
+* VM and/or Bare metal instances are provisioned for the workloads and we are able to login to console.
+* SkyWalking instance has been setup either on premises or in Hyperscalers. Ensure that inbound ingress traffic is allowed on port 11800 for SkyWalking instance.
+
+## **Steps to configure monitoring**
+
+### Step 1- Monitoring host metric with node exporter
+
+Login to the console of machine to be monitored.
+
+Next, Download and run the node exporter using below code snippet
+
+```shell
+wget https://github.com/prometheus/node_exporter/releases/download/v1.4.0-rc.0/node_exporter-1.4.0-rc.0.linux-amd64.tar.gz
+
+tar xvfz node_exporter-1.4.0-rc.0.linux-amd64.tar.gz
+
+rm -rf node_exporter-1.4.0-rc.0.linux-amd64.tar.gz
+
+cd node_exporter-1.4.0-rc.0.linux-amd64
+
+cp node_exporter /usr/sbin
+
+cat<<EOF >nodeexporter.service
+
+[Unit]
+Description=Service to start the node_exporter on the node
+[Service]
+Type=Simple
+ExecStart=/usr/sbin/node_exporter
+StandardOutput=syslog
+StandardError=syslog
+SyslogIdentifier=%n
+
+[Install]
+WantedBy=multi-user.target
+
+EOF
+
+
+mv nodeexporter.service /etc/systemd/system
+systemctl daemon-reload
+systemctl enable nodexporter.service
+systemctl start nodexporter.service
+```
+
+Check the metrics endpoint shows metrics exported from the infrastructure node.
+
+```shell
+curl <http://localhost:9100/metrics>
+```
+
+![](/img/node_metrics.png "Node Metrics")
+
+### Step 2- Setup OTel (OpenTelemetry) collector
+
+OTel provides a set of standardized vendor-agnostic SDKs, APIs, and tools for ingesting, transforming, and sending data to an Observability back-end. We can install OTel collector in the same host, which exposes metrics, or we can create central VM/Bare metal instance to receive telemetry information from several infrastructure instances.
+
+Install OTel collector.
+
+```shell
+sudo wget <https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.54.0/otelcol_0.54.0_linux_amd64.deb>
+
+sudo dpkg -i otelcol_0.54.0_linux_amd64.deb
+
+systemctl status otelcol
+
+# By default, the otelcol systemd service will be started with the --config=/etc/otelcol/config.yaml option after installation.
+```
+
+Create a new otel configuration file (say /etc/otelcol/otel-collector-config.yaml) to link the metrics from host nodes to SkyWalking oap. The orange box in below images must be modified to point to correct infrastructure hosts and SkyWalking DNS or IP address.
+
+![](/img/otel_collector_configuration.png "OTeL Configuration")
+
+Change OTel configuration to newly created OTel configuration file.
+
+```shell
+cd /etc/otelcol
+
+sudo nano otelcol.conf
+
+#change the path to new configuration file we created in the above step and save
```

Comment:
Hmm... should this be text and actually read "Change the path to the new configuration file you just created in the above step and save." (complete with the period)? I generally don't ask folks to change things in their code blocks (that's Denis' job :-)  but I'm wondering if this is something that really doesn't belong in the code block.